File: CPP/Leet Code Solutions/palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }
        int reverse = 0;
        int original = x;
        while(x!=0){
            int remainder = x%10;
            reverse = reverse*10 + remainder;
            x = x/10;
        }
        return reverse==original;
    }
}

File: CPP/Leet Code Solutions/palindromeNumber.cpp

class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return 0;
        if(x == 0) return 1;
        long long st = x, en = 0;
        while(x > 0){
            en = en*10 + x%10;
            x/=10;
        }
        if(en == st) return 1;
        else return 0;
    }
};
